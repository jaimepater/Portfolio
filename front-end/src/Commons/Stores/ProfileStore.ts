import { action, computed, observable } from 'mobx';
import moment, { Moment } from 'moment';
import { profileRef } from '../Firebase/Firebase';

const NOW = 'Now';
const FORMAT = 'MMM YYYY';

export interface IProfileStore {
  profileData: IPortfolioData;
  getProfileData: any;
  fullName: any;
  experienceTitle: any;
  profiles: any;
  experienceList: any;
}

export interface IPortfolioData {
  profile?: {
    firstName: string;
    lastName: string;
    github: string;
    strava: string;
    mail: string;
    linkedin: string;
    list: Array<string>;
  };
  experience?: {
    title: string;
    list: Array<IExperienceList>;
  };
  skills?: {
    title: string;
    list: Array<ISkill>;
  };
  projects?: {
    title: string;
    list: Array<IProject>;
  };
  hobbies?: {
    title: string;
    list: Array<Hobbie>;
  };
}

export interface ISkill {
  id: string;
  name: string;
  score: string;
}

export interface IExperienceList {
  title: string;
  id: string;
  color: string;
  responsibilities: Array<string>;
  from: {
    display: string;
    date: Moment;
  };
  to: {
    display: string;
    date: Moment;
  };
  description: string;
  technologies: Array<string>;
}

export interface IProject {
  title: string;
  color: string;
  description: string;
  github: string;
  technologies: Array<string>;
  mainTechnologies: Array<string>;
}

export interface Hobbie {
  title: string;
  color: string;
  logo: string;
  description: string;
}

export default class ProfileStore implements IProfileStore {
  @observable profileData: IPortfolioData = {};

  @computed get fullName() {
    let name = '';
    if (this.profileData.profile) {
      name = `${this.profileData!.profile!.firstName} ${this.profileData!.profile!.lastName}`;
    }
    return name;
  }

  @computed get profile() {
    let profile;
    if (this.profileData.profile) {
      profile = this.profileData.profile;
    }
    return profile;
  }

  @computed get profiles() {
    let profiles: Array<string> = [];
    if (this.profileData.profile) {
      profiles = this.profileData.profile.list;
    }
    return profiles;
  }

  @computed get experienceTitle() {
    let experience;
    if (this.profileData.experience) {
      experience = this.profileData.experience.title;
    }
    return experience;
  }

  @computed get skillTitle() {
    let experience;
    if (this.profileData.skills) {
      experience = this.profileData.skills.title;
    }
    return experience;
  }

  @computed get skillsList() {
    let experience;
    if (this.profileData.skills) {
      experience = this.profileData.skills.list;
    }
    return experience;
  }

  @computed get projectsTitle() {
    let projects;
    if (this.profileData.projects) {
      projects = this.profileData.projects.title;
    }
    return projects;
  }

  @computed get projectsList() {
    let projects;
    if (this.profileData.projects) {
      projects = this.profileData.projects.list;
    }
    return projects;
  }

  @computed get hobbiesTitle() {
    let projects;
    if (this.profileData.hobbies) {
      projects = this.profileData.hobbies.title;
    }
    return projects;
  }

  @computed get hobbiesList() {
    let projects;
    if (this.profileData.hobbies) {
      projects = this.profileData.hobbies.list;
    }
    return projects;
  }

  @computed get experienceList() {
    let experienceList;
    if (this.profileData.experience) {
      experienceList = this.profileData.experience.list.map(item => ({
        ...item,
        from: this.getDate(item.from),
        to: this.getDate(item.to),
      }));
    }
    return experienceList;
  }

  getDate = (stringDate: any) => {
    if (stringDate === NOW) {
      return { date: moment(), display: NOW };
    }
    return {
      date: moment(stringDate),
      display: moment(stringDate).format(FORMAT),
    };
  };

  @action
  public getProfileData() {
    return profileRef.on('value', snapshot => {
      this.profileData = snapshot.val();
    });
  }
}
