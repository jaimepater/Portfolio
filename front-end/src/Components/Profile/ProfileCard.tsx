import * as React from 'react';
import { Card, CardContent } from '@material-ui/core';
import styled from 'styled-components';
import { observer } from 'mobx-react';
import { ReactNode } from 'react';
import ProfileName from './ProfileName';
import ProfileList from './ProfileList';

export const StyledCard = styled(Card)`
  margin-top: -50px;
`;

const StyledCardContent = styled(CardContent)`
  padding-top: 55px;
`;

interface ProfileCardProps {
  name: string;
  flag?: ReactNode;
  list: string[];
  time?: number;
}

const ProfileCard = observer(({ name, flag, list, time = 5000 }: ProfileCardProps) => {
  return (
    <StyledCard>
      <StyledCardContent>
        <div>
          <ProfileName name={name} flag={flag} />
          <ProfileList profiles={list || []} time={time} />
        </div>
      </StyledCardContent>
    </StyledCard>
  );
});

export default ProfileCard;
