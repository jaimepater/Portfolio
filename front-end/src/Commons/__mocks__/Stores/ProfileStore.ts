import {IPortfolioData, IProfileStore} from "../../Stores/ProfileStore";
import portfolioData from "./profileData.json";
// / <reference types="react-scripts" />

export default class ProfileStoreMock implements IProfileStore {
  profileData : IPortfolioData = portfolioData ;

  fullName: jest.Mocked<any>;

  profiles: jest.Mocked<any>;

  getProfileData: jest.Mocked<any>;

  experienceTitle: jest.Mocked<any>;
}
