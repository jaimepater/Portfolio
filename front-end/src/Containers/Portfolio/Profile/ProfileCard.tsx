// @flow
import * as React from 'react';
import {
  Card, CardActionArea, CardContent, Typography,
} from '@material-ui/core';
import styled from 'styled-components';
import { useEffect, useState } from 'react';
import { inject, observer } from 'mobx-react';
import { IProfileStore } from '../../../Commons/Stores/ProfileStore';
import ProfileName from './ProfileName';


const StyledCard = styled(Card)`
  margin-top: -50px;
`;

const StyledCardContent = styled(CardContent)`
  padding-top: 55px;
`;

interface CardProps {
  profileStore? : IProfileStore
}
const ProfileCard: React.FC<CardProps> = ({ profileStore } : CardProps) => {
  useEffect(() => {
    console.log('hiiiiiii');
    profileStore && profileStore.getProfileData();
    console.log(profileStore && profileStore.profileData.name);
  }, [profileStore]);


  return (
    <StyledCard>
      <StyledCardContent>
        <div>
          <ProfileName name={profileStore && profileStore.profileData.name} />
        </div>
      </StyledCardContent>
    </StyledCard>
  );
};

export default inject('profileStore')(observer(ProfileCard));
