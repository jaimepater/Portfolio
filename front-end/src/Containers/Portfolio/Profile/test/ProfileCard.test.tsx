import React from "react";
import {shallow} from 'enzyme';
import ProfileCard from "../ProfileCard";
import useStores from "../../../../Commons/Hooks/Stores";
import ProfileStoreMock from "../../../../Commons/__mocks__/Stores/ProfileStore";
import ProfileStore from "../../../../Commons/Stores/ProfileStore";





describe('ProfileCard', () => {
  it('Renders ProfileCard', () => {
    const shallowComponent = shallow(<ProfileCard />);
    expect(shallowComponent).toMatchSnapshot();
  });
});