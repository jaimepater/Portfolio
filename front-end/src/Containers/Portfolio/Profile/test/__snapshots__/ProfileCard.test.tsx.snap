// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCard Renders ProfileCard 1`] = `
<Styled(WithStyles(ForwardRef(Card)))>
  <Styled(WithStyles(ForwardRef(CardContent)))>
    <div>
      <ProfileName
        name=""
      />
      <ProfileList
        profiles={Array []}
        time={5000}
      />
    </div>
  </Styled(WithStyles(ForwardRef(CardContent)))>
</Styled(WithStyles(ForwardRef(Card)))>
`;
